### PR TITLE
feat(workspace): desktop support update network

### PR DESCRIPTION
**Pull request by wuzhuanhong**
**What this PR does / why we need it**:

Desktop support update network

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST=./huaweicloud/services/acceptance/workspace TESTARGS='-run TestAccDesktop_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/workspace -v -run TestAccDesktop_basic -timeout 360m -parallel 4
=== RUN   TestAccDesktop_basic
=== PAUSE TestAccDesktop_basic
=== CONT  TestAccDesktop_basic
--- PASS: TestAccDesktop_basic (1599.58s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/workspace 1599.630s```

### CODE REVIEW

File: huaweicloud/services/workspace/resource_huaweicloud_workspace_desktop.go [hunk 1]
```diff
@@ -388,6 +386,25 @@ func flattenDesktopSecurityGroups(securityGroups []desktops.SecurityGroup) []int
...
+		return nil, err
+	}
+
+	nic := []map[string]interface{}{
+		{
+			"network_id": network[0].Subnet.ID,
+		},
+	}
+
+	return nic, err
```
Comment (Lance52259) [inline huaweicloud/services/workspace/resource_huaweicloud_workspace_desktop.go]:
```suggestion
	if len(network) < 1 {
		return nil, fmt.Errorf("unable to find any network information under Workspace desktop (%s)", desktopId)
	}

	return []map[string]interface{}{
		{
			"network_id": network[0].Subnet.ID,
		},
	}, nil
```

File: huaweicloud/services/workspace/resource_huaweicloud_workspace_desktop.go [hunk 2]
```diff
@@ -548,6 +573,40 @@ func updateDesktopVolumes(ctx context.Context, client *golangsdk.ServiceClient,
...
+	opts := desktops.UpdateNetworkOpts{
+		DesktopId:        desktopId,
+		VpcId:            d.Get("vpc_id").(string),
+		SubnetId:         nicVal["network_id"].(string),
+		SecurityGroupIds: listId,
+	}
+
+	resp, err := desktops.UpdateNetwork(client, opts)
+	if err != nil {
+		return fmt.Errorf("error update network Workspace desktop: %s", err)
```
Comment (Lance52259) [inline huaweicloud/services/workspace/resource_huaweicloud_workspace_desktop.go]:
```suggestion
		return fmt.Errorf("error updating the network of the Workspace desktop (%s): %s", desktopId, err)
```

File: huaweicloud/services/workspace/resource_huaweicloud_workspace_desktop.go (hunk 1 above)
Comment (wuzhuanhong) [reply huaweicloud/services/workspace/resource_huaweicloud_workspace_desktop.go]: fixed

File: huaweicloud/services/workspace/resource_huaweicloud_workspace_desktop.go (hunk 2 above)
Comment (wuzhuanhong) [reply huaweicloud/services/workspace/resource_huaweicloud_workspace_desktop.go]: fixed

**Comment by wuzhuanhong** [comment]
make testacc TEST=./huaweicloud/services/acceptance/workspace TESTARGS='-run TestAccDesktop_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/workspace -v -run TestAccDesktop_basic -timeout 360m -parallel 4
=== RUN   TestAccDesktop_basic
=== PAUSE TestAccDesktop_basic
=== CONT  TestAccDesktop_basic
--- PASS: TestAccDesktop_basic (1531.89s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/workspace 1531.941s

File: huaweicloud/services/workspace/resource_huaweicloud_workspace_desktop.go [hunk 3]
```diff
@@ -426,6 +445,13 @@ func resourceDesktopRead(_ context.Context, d *schema.ResourceData, meta interfa
 		mErr = multierror.Append(mErr, d.Set("security_groups", flattenDesktopSecurityGroups(securityGroups)))
 	}
 
+	nicVal, err := getDesktopNetwork(client, desktopId)
+	if err != nil {
+		mErr = multierror.Append(mErr, err)
+	} else {
+		mErr = multierror.Append(nil, d.Set("nic", nicVal))
```
Comment (ShiChangkuo) [inline huaweicloud/services/workspace/resource_huaweicloud_workspace_desktop.go]:
```suggestion
		mErr = multierror.Append(mErr, d.Set("nic", nicVal))
```

Comment (wuzhuanhong) [reply huaweicloud/services/workspace/resource_huaweicloud_workspace_desktop.go]: fixed

**Comment by wuzhuanhong** [comment]
make testacc TEST=./huaweicloud/services/acceptance/workspace TESTARGS='-run TestAccDesktop_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/workspace -v -run TestAccDesktop_basic -timeout 360m -parallel 4
=== RUN   TestAccDesktop_basic
=== PAUSE TestAccDesktop_basic
=== CONT  TestAccDesktop_basic
--- PASS: TestAccDesktop_basic (1681.44s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/workspace 1681.504s

**Pull request merged**